src/infill/ZigzagConnectorProcessorEndPieces.cpp

```diff
@@ -11,11 +11,6 @@ void ZigzagConnectorProcessorEndPieces::registerVertex(const Point& vertex)
     {
         first_zigzag_connector.push_back(vertex);
     }
-    else if (last_scanline_is_even)
-    { // when a boundary segments starts in an even scanline it's either a normal zigzag connector or an endpiece to be included
-        // note that for ZigzagConnectorProcessorDisconnectedEndPieces only the last line segment from a boundary vertex to a scanline-boundary intersection is omitted
-        addLine(last_connector_point, vertex);
```

Comment:
Fixed in 34be16e32c495e7e276bf17c3636f32419a42836 and 976dd9a1ceb69c3f00cc23437d2fa343e4277cab